sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -118,6 +119,12 @@ private[sql] class SharedState(
     statusStore
   }
 
+  sparkContext.statusStore.diskStore.foreach { kvStore =>
+    sparkContext.listenerBus.addToQueue(
+      new DiagnosticListener(conf, kvStore.asInstanceOf[ElementTrackingStore]),
```

Comment:
Why do we need to share the same kvStore?

Reply:
@dongjoon-hyun 
I used the same way that the in-memory kvStore used. which is:
1. define a shared kvStore in SparkContext
2. share the kvStore with the listeners that need to update the store (Jobs UI Listener, SQL UI Listener)
3. share the kvStore with the components that need to read the store (web UI, rest API)